pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/ScalarTransformFunctionWrapperTest.java

```diff
@@ -802,6 +802,57 @@ public void testArrayConcatIntTransformFunction() {
     testTransformFunctionMV(transformFunction, expectedValues);
   }
 
+  @Test
```

Comment:
Please also consider adding some other query execution tests. Take a look at `BaseMultiValueQueriesTest` or `OfflineClusterIntegrationTest` to add such tests.

Example cases ...

**Nest the function**

```
SELECT arraylength(arrayConcatInt(mvIntCol1, mvIntCol2))
FROM FOO
```

```
SELECT arraySortInt(arrayConcatInt(mvIntCol1, mvIntCol2))
FROM FOO
```

**Use the output of the expression in WHERE clause**

```
SELECT arrayConcatInt(mvIntCol1, mvIntCol2) AS concatenated
FROM FOO
WHERE arrayLength(concatenated) >= 2
```

**Use in GROUP BY**

```
SELECT arrayConcatInt(mvIntCol1, mvIntCol2) AS concatenated, sum(intcol)
FROM FOO
WHERE arrayLength(concatenated) >= 2
GROUP BY concatenated
```

Reply:
Sidd:

Sure this is a good idea, just to confirm: such tests do not exists for the currently supported functions like arrayConcatInt and arrayConcatString, right? 

If that is the case I can look into adding these tests for the new functions being added in this PR, I can file a new issue to support those that were added earlier so we don't hold this on earlier tech debt.

Reply:
> such tests do not exists for the currently supported functions like arrayConcatInt and arrayConcatString, right?

I think that is correct. But we will have to dig through since tests are spread across so many files unfortunately.

> If that is the case I can look into adding these tests for the new functions being added in this PR

Yes, in the PR I suggest adding tests only for new functions. Separate PR can be filed to add more tests for existing functions.